gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java

```diff
@@ -73,25 +81,47 @@ public GroovyCodeEditor(Scene scene) {
         codeArea.richChanges()
                 .filter(ch -> !ch.getInserted().equals(ch.getRemoved()))
                 .subscribe(change -> codeArea.setStyleSpans(0, computeHighlighting(codeArea.getText())));
-        SearchBar searchBar = new SearchBar(codeArea);
+        searchBar = new SearchBar(codeArea);
         searchBar.setCloseAction(e -> {
             setShowDetailNode(false);
             codeArea.requestFocus();
         });
+        replaceWordBar = new ReplaceWordBar();
         setMasterNode(new VirtualizedScrollPane(codeArea));
-        setDetailNode(searchBar);
+        VBox vBox = new VBox();
+        vBox.getChildren().addAll(searchBar, replaceWordBar);
+        vBox.setSpacing(7);
+        setDetailNode(vBox);
         setDetailSide(Side.TOP);
         setShowDetailNode(false);
 
         setOnKeyPressed((KeyEvent ke) -> {
             if (searchKeyCombination.match(ke)) {
+                setShowDetailNode(false);
+                vBox.getChildren().remove(replaceWordBar);
+                setDetailNode(vBox);
+                resetDividerPosition();
                 if (codeArea.getSelectedText() != null && !"".equals(codeArea.getSelectedText())) {
                     searchBar.setSearchPattern(codeArea.getSelectedText());
                 }
-                if (!isShowDetailNode()) {
-                    setShowDetailNode(true);
+                showDetailNode();
+                vBox.requestFocus();
+            } else if (replaceWordKeyCombination.match(ke)) {
+                setShowDetailNode(false);
+                if (!vBox.getChildren().contains(replaceWordBar)) {
+                    vBox.getChildren().add(replaceWordBar);
+                }
+                setDetailNode(vBox);
+                resetDividerPosition();
+                replaceWordBar.getReplaceAllButton().disableProperty().bind(validateDisableProperty());
+                replaceWordBar.getReplaceButton().disableProperty().bind(validateDisableProperty());
+                replaceWordBar.getReplaceAllButton().setOnAction(event -> replaceAllOccurences(searchBar.getSearchField().getText(), codeArea.getText()));
+                replaceWordBar.getReplaceButton().setOnAction(event -> replaceCurrentOccurence(searchBar.getMatcherCurrentMatchStart(), searchBar.getMatcherCurrentMatchEnd()));
```

Comment:
There is a design issue here with the encapsulation principle: you should not be able to disable the buttons like this. This should be done internally in the replaceBar

Reply:
updated